manifests/networkd.pp

```diff
@@ -5,10 +5,53 @@
 # @param ensure The state that the ``networkd`` service should be in
 # @param path path where all networkd files are placed in
 # @param manage_all_network_files if enabled, all networkd files that aren't managed by puppet will be purged
+# @param link_profiles
+#   Hash of network link profiles that can be referenced by its key on an interface
+#   The structure is equal to the 'link' parameter of an interface.
+# @param netdev_profiles
+#   Hash of netdev profiles that can be referenced by its key on an interface
+#   The structure is equal to the 'netdev' parameter of an interface.
+# @param network_profiles
+#   Hash of network profiles that can be referenced by its key on an interface
+#   The structure is equal to the 'network' parameter of an interface.
+# @param interfaces
+#  Hash of interfaces to configure on a node.
+#  The link, netdev and network parameters are deep merged with the respective profile
+#  (referenced by the key of the interface).
+#  With the profiles you can set the default values for a network.
+#  Hint: to remove a profile setting for an interface you can either overwrite or
+#  set it to `~` for removal.
+#  Example (hiera yaml notation):
+#    systemd::networkd::network_profiles:
+#      mynet:
+#        Network:
+#          Gateway: '192.168.0.1'
+#        
+#    systemd::networkd
+#      mynet:
+#        filename: '50-static'
+#        network:
+#          Match:
+#            Name: 'eno0'
+#          Network:
+#            Address: '192.168.0.15/24'
+#
+#   Gives you a file /etc/systemd/network/50-static.network
+#   with content:
+#     [Match]
+#     Name=enp2s0
+#     [Network]
+#     Address=192.168.0.15/24
+#     Gateway=192.168.0.1
+#
```

Comment:
Maybe this empty line is breaking puppet-strings

```suggestion
```

Reply:
no it does not. systemd::networkd is set to a private api, this is why none of the parameters show up in the REFERENCE.md file ;(
I think to make them visible we have to either remove the private api sign or add the parameters to init.pp (Both of them are not verry good in my opinion)

Reply:
Oh OK, I thought private classes were still documented. I guess that's fine then.